docs(framework): Update quickstart-docker guide to use SuperExec (#5737)
adap/flower: framework/docs/source/docker/tutorial-quickstart-docker.rst

    * ``docker run``: This tells Docker to run a container from an image.
    * ``--rm``: Remove the container once it is stopped or the command exits.
    * ``-p 9091:9091 -p 9092:9092 -p 9093:9093``: Map port ``9091``, ``9092`` and ``9093`` of the
      container to the same port of the host machine, allowing other services to access the
      ServerAppIO API on ``http://localhost:9091``, the Fleet API on ``http://localhost:9092`` and
      the Control API on ``http://localhost:9093``.
    * ``--network flwr-network``: Make the container join the network named ``flwr-network``.
    * ``--name superlink``: Assign the name ``superlink`` to the container.
    * ``--detach``: Run the container in the background, freeing up the terminal.
    * :substitution-code:`flwr/superlink:|stable_flwr_version|`: The name of the image to be run and the specific
      tag of the image. The tag :substitution-code:`|stable_flwr_version|` represents a :doc:`specific version <pin-version>` of the image.
    * ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
      unencrypted communication.

>     * ``--isolation process``: Tells the SuperLink that the ServerApp is created by separate
>       independent process. The SuperLink does not attempt to create it. You can learn more about
>       the different process modes here: :doc:`run-as-subprocess`.

Reviewer: ```suggestion
    * ``--isolation process``: Tells the SuperLink that the ServerApp is executed by separate
      independent process. The SuperLink does not attempt to execute it. You can learn more about
      the different process modes here: :doc:`run-as-subprocess`.
```

       * ``--rm``: Remove the container once it is stopped or the command exits.
       * ``-p 9094:9094``: Map port ``9094`` of the container to the same port of
         the host machine, allowing other services to access the SuperNode API on
         ``http://localhost:9094``.
       * ``--network flwr-network``: Make the container join the network named ``flwr-network``.
       * ``--name supernode-1``: Assign the name ``supernode-1`` to the container.
       * ``--detach``: Run the container in the background, freeing up the terminal.
       * :substitution-code:`flwr/supernode:|stable_flwr_version|`: This is the name of the
         image to be run and the specific tag of the image.
       * ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
         unencrypted communication.
       * ``--superlink superlink:9092``: Connect to the SuperLink's Fleet API at the address
         ``superlink:9092``.
       * ``--node-config "partition-id=0 num-partitions=2"``: Set the partition ID to ``0`` and the
         number of partitions to ``2`` for the SuperNode configuration.

>        * ``--clientappio-api-address 0.0.0.0:9094``: Set the address and port number that the
>          SuperNode is listening on to communicate with the ClientApp. If
>          two SuperNodes are started on the same machine, set two different port numbers for each SuperNode.
>          (E.g. In the next step, we set the second SuperNode container to listen on port 9095)

Reviewer: ```suggestion
       * ``--clientappio-api-address 0.0.0.0:9094``: Set the address and port number that the
         SuperNode is listening on to communicate with the ClientApp process. If
         two SuperNodes are started on the same machine, use two different port numbers for each SuperNode.
         (E.g. In the next step, we set the second SuperNode container to listen on port 9095)
```

           --clientappio-api-address 0.0.0.0:9095 \
           --isolation process

Step 4: Start the SuperExec for ServerApps
------------------------------------------

The **SuperExec** Docker image comes with a pre-installed version of Flower and serves
as a base for building your own image. Use a **single** image and select the desired
behavior (ServerApps vs ClientApps) at runtime via the ``--plugin-type`` flag. SuperExec
spawns the corresponding processes on demand.

When using SuperExec with the *serverapp* plugin, pass ``--appio-api-address`` pointing
to the SuperLink's **ServerAppIO API** endpoint.

1. Create a Dockerfile called ``superexec.Dockerfile`` and paste the following:

   .. code-block:: dockerfile
       :caption: superexec.Dockerfile
       :substitutions:

>        FROM flwr/superexec:|stable_flwr_version|

Reviewer 2: ```suggestion
       FROM flwr/superexec:1.21.0.dev20250819
```

I suggest setting it to a `dev` tag for now because we did not release the SuperExec image for `1.20.0`.

Author: Thanks for bringing this up. Yes, we also notice this error and will fix it.

Reviewer 2: Resolving comment - addressed by #5740.

          && python -m pip install -U --no-cache-dir .

       ENTRYPOINT ["flower-superexec"]

   .. dropdown:: Understand the Dockerfile

       * :substitution-code:`FROM flwr/superexec:|stable_flwr_version|`: This line specifies that the Docker image
         to be built from is the ``flwr/superexec`` image, version :substitution-code:`|stable_flwr_version|`.
       * ``WORKDIR /app``: Set the working directory for the container to ``/app``.
         Any subsequent commands that reference a directory will be relative to this directory.
       * ``COPY pyproject.toml .``: Copy the ``pyproject.toml`` file.
         from the current working directory into the container's ``/app`` directory.
       * ``RUN sed -i 's/.*flwr\[simulation\].*//' pyproject.toml``: Remove the ``flwr`` dependency
         from the ``pyproject.toml``.
       * ``python -m pip install -U --no-cache-dir .``: Run the ``pip`` install command to
         install the dependencies defined in the ``pyproject.toml`` file.

         The ``-U`` flag indicates that any existing packages should be upgraded, and
         ``--no-cache-dir`` prevents pip from using the cache to speed up the installation.

>        * ``ENTRYPOINT ["flwr-superexec"]``: Set the command ``flwr-superexec`` to be

Author: Great catch! Thanks.
